File: 2506. Count Pairs Of Similar Strings/Solution.py
from collections import defaultdict


class Solution:
    def similarPairs(self, words: list[str]) -> int:
        cnt = defaultdict(int)
        ans = 0
        for word in words:
            key = 0
            for c in word:
                key |= 1 << (ord(c) - ord("a"))
            ans += cnt[key]
            cnt[key] += 1
        return ans

File: 2506. Count Pairs Of Similar Strings/Solution.cpp
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    int similarPairs(vector<string>& words)
    {
        unordered_map<unsigned, int> cnt;
        int ans = 0;
        for (const string& word : words) {
            unsigned key = 0;
            for (char c : word)
                key |= 1 << (c - 'a');
            ans += cnt[key]++;
        }
        return ans;
    }
};
